lablueprint/friends-of-the-children
# Profile bug and password

Bug fixes:
- Edited profile fields to match the figure, removed extra fields
- Previously all users could update their profile to change their role but they shouldn't be able to, now they can't change their role

New password functionality:
- added old password and new password fields
- user needs to enter old password (and verify it matches) in order to be able to edit and update new password, old password is checked using bcrypt to compare the hashes
- after the old password is verified, user can update the new password by typing it in and saving profile --> the bcrypt hashed password will be uploaded to firebase
- the password fields can toggle visibility with an eye icon
- the bio field can also be updated

Testing:
- check that the bio will be updated
- check that the old password is correctly verified
- check that the new password is correctly updated
- check that the fields match what is in the figma (except the password fields which might not exist in figma yet)
